Yoast/wordpress-seo
# Duplicate logic from WP Core to filter pages before preloading them a…

…nd their indexables

## Context


* WP_Query when on the page overview loads every single page ( but only their ID and parent_id ). This caused our preloading logic to load all data and indexables for every single page.

## Summary


This PR can be summarized in the following changelog entry:

* Fixes an issue where memory could be exhaused on the page overview for users with large amounts of pages.

## Relevant technical choices:

* Two functions from WP Core had to be duplicated with slight adaptations in order to ensure we're filtering the list of all page IDs down to the list of currently shown page IDs in the exact same manner.
* The CS warning threshold has been increased by 3 as the 3 warnings, class name too many parts and `@inheritDoc` comments not having short descriptions should not be warnings.

## Test instructions

This PR can be tested by following these steps:

* Make sure you have more pages than fit on a single overview in your admin.
* Make sure this includes child pages that are shown on the first page of your pages overview.
* Open the first page of your pages overview.
* Open query monitor.
* There should be a single query grabbing all indexables for pages on your current overview.
* No additional indexables should be fetched.
* There should be no additional queries for other page indexables.

## UI changes

## Documentation

## Quality assurance


Fixes https://yoast.atlassian.net/browse/QAK-2234
Fixes https://github.com/Yoast/wordpress-seo/issues/16027
